app/presenters/any_link_presenter.rb

```diff
@@ -0,0 +1,92 @@
+class AnyLinkPresenter
+  def self.for(link_container, *args)
+    presenter_klass =
+      case link_container
+      when Link
+        ForALink
+      when LocalAuthority
+        ForALocalAuthority
+      end
+    presenter_klass.new(link_container, *args)
+  end
+
+  class ForALink < SimpleDelegator
+    include UrlStatusPresentation
+    attr_reader :with_service_links_count
+
+    def initialize(link, with_service: nil, with_service_links_count: 0, context:, local_authority: nil)
+      super(link)
+      @with_service = with_service
+      @with_service_links_count = with_service_links_count
+      @context = context
```

Comment:
Is there a better name for `context`? It's not immediately clear to me what that refers to.